cmd/lightclient/sentinel/handlers.go

```diff
@@ -47,6 +70,56 @@ func pingHandler(stream network.Stream) {
 	}
 }
 
+func statusHandler(stream network.Stream) {
+	setDeadLines(stream)
+
+	log.Info("Got status request")
+}
+
+func goodbyeHandler(stream network.Stream) {
+	goodByeBytes := make([]byte, 8)
+	setDeadLines(stream)
+	n, err := stream.Read(goodByeBytes)
+	if err != nil {
+		log.Warn("Goodbye handler crashed", "err", err)
+		return
+	}
+
+	if n != 8 {
+		log.Warn("Invalid goodbye message received")
+		return
+	}
+
+	log.Info("[Lightclient] Received", "goodbye", ssz.UnmarshallUint64(goodByeBytes))
+	n, err = stream.Write(goodByeBytes)
+	if err != nil {
+		log.Warn("Goodbye handler crashed", "err", err)
+		return
+	}
+
+	if n != 8 {
+		log.Warn("Could not send Goodbye")
+	}
+
+	peerId := stream.Conn().RemotePeer()
+	disconnectPeerCh <- peerId
```

Comment:
again I do not like this, this is asking for troubles, possibility of deadlocks.